File: backend/src/main/java/com/serendipity/backend/model/entity/Utente.java
package com.serendipity.backend.model.entity;

import com.serendipity.backend.model.enums.Ruolo;
import jakarta.persistence.*;

@Entity
public class Utente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String codiceFiscale;

    private String nome;

    private String cognome;

    private String email;

    private String password;

    @Enumerated(EnumType.STRING)
    private Ruolo ruolo;

    @Column(name = "paga_oraria")
    private Double pagaOraria;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Ruolo getRuolo() {
        return ruolo;
    }

    public void setRuolo(Ruolo ruolo) {
        this.ruolo = ruolo;
    }

    public Double getPagaOraria() {
        return pagaOraria;
    }

    public void setPagaOraria(Double pagaOraria) {
        this.pagaOraria = pagaOraria;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getCognome() {
        return cognome;
    }

    public void setCognome(String cognome) {
        this.cognome = cognome;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getCodiceFiscale() {
        return codiceFiscale;
    }

    public void setCodiceFiscale(String codiceFiscale) {
        this.codiceFiscale = codiceFiscale;
    }

    public Utente() {
    }

    public Utente(Long id, String codiceFiscale, String nome, String cognome, String email, String password, Ruolo ruolo, Double pagaOraria) {
        this.id = id;
        this.codiceFiscale = codiceFiscale;
        this.nome = nome;
        this.cognome = cognome;
        this.email = email;
        this.password = password;
        this.ruolo = ruolo;
        this.pagaOraria = pagaOraria;
    }
}

File: frontend/src/types/utente.ts
export type Ruolo = 'ADMIN' | 'DIPENDENTE';

export interface UtenteDto {
    id: number;
    codiceFiscale: string;
    nome: string;
    cognome: string;
    email: string;
    ruolo: Ruolo;
    pagaOraria: number | null;
}

export interface CreaUtenteDto {
    codiceFiscale: string;
    nome: string;
    cognome: string;
    email: string;
    password: string;
    ruolo: Ruolo;
    pagaOraria?: number | null;
}

export interface AggiornaPasswordDto {
    oldPassword: string;
    newPassword: string;
}

export interface ProfiloUtenteDto {
    id: number;
    codiceFiscale: string;
    nome: string;
    cognome: string;
    email: string;
    ruolo: Ruolo;
}
